### PR TITLE
Add circleci configuration

### PR DESCRIPTION
**Pull request by EdJoJob**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
+---
+version: 2.1
+executors:
+  med: # 2cpu, 4G ram
+    docker:
+      - image: circleci/node:10
+    resource_class: medium
+    working_directory: ~/project
+    environment:
+      GRADLE_OPTS: -Xmx2048m -Dorg.gradle.daemon=false -Dorg.gradle.parallel=true -Dorg.gradle.workers.max=2
```
Comment (joshuafernandes) [inline .circleci/config.yml]: Do we need this line for a node build? Or are there java deps in the tests which will use this?

Comment (EdJoJob) [reply .circleci/config.yml]: Just bad copy-paste, now fixed

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  prepare:
+    description: "Prepare"
+    steps:
+      - checkout
+      - restore_cache:
+          name: Restore cached dependencies
+          keys:
+            - deps-{{ checksum "package.json" }}-{{ .Branch }}-{{ .Revision }}
+            - deps-{{ checksum "package.json" }}
+            - deps-
```
Comment (NicolasMassart) [inline .circleci/config.yml]: are you sure about this last key?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            mv test-results/test-contract-results.xml test-results/contracts/results.xml
+      - store_test_results:
+          path: test-results
+      - run:
+          name: Collect Contract Coverage
+          command: |
+            yarn run coverage:contracts
+            ./node_modules/.bin/istanbul report cobertura --root .
+      - store_artifacts:
+          path: coverage
```
Comment (NicolasMassart) [inline .circleci/config.yml]: does this 1st artifact exist? Is it a directory?

Comment (EdJoJob) [reply .circleci/config.yml]: This "artifact" is a directory

File: .circleci/config.yml (hunk 2 above)
Comment (EdJoJob) [reply .circleci/config.yml]: Yes, this allows for most of the cache to be restored in the case where a single dependency has been updated.
In most cases we will be able to use most of the cache even in the case where the `package.json` file has been changed.

Comment (EdJoJob) [reply .circleci/config.yml]: See https://circleci.com/docs/2.0/caching/#restoring-cache for more details

**Pull request merged**